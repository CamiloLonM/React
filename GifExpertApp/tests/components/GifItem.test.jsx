import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Test is <GifItem /> ', () => {
  const title = 'Dindon';
  const url = 'https://dindon-dam.com/dindon.jpg';

  test('You must match the snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('It should display the image with the URL and Alt indicated', () => {
    render(<GifItem title={title} url={url} />);

    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test('You must display the title in the component', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
