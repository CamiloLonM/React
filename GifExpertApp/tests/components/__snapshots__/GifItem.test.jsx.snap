// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test is <GifItem />  You must match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dindon"
      src="https://dindon-dam.com/dindon.jpg"
    />
    <p>
      Dindon
    </p>
  </div>
</div>
`;
